swift/apple/FirezoneNetworkExtension/Adapter.swift

```
@@ -317,19 +316,13 @@ class Adapter: @unchecked Sendable {
       let ipv4, let ipv6, let dns, let searchDomain, let ipv4Routes, let ipv6Routes):
       Log.log("Received TunInterfaceUpdated event")
 
-      // Decode all data into local variables first to ensure all parsing succeeds before applying
-      guard let dnsData = dns.data(using: .utf8),
-        let dnsAddresses = try? JSONDecoder().decode([String].self, from: dnsData),
-        let data4 = ipv4Routes.data(using: .utf8),
-        let data6 = ipv6Routes.data(using: .utf8),
-        let decoded4 = try? JSONDecoder().decode([NetworkSettings.Cidr].self, from: data4),
-        let decoded6 = try? JSONDecoder().decode([NetworkSettings.Cidr].self, from: data6)
-      else {
-        fatalError("Could not decode network configuration from connlib")
+      // Convert UniFFI types to NetworkExtension types
+      let routes4 = ipv4Routes.compactMap { cidr in
```

`compactMap` skips nil I guess? It would be good to not silently drop these but add a warning instead.

---

If we throw an error instead of returning nil from the extension, we can do a try-catch here, log the error properly and return `nil` to still skip it.